hail/src/main/scala/is/hail/io/RowStore.scala

```diff
@@ -15,13 +15,14 @@ import is.hail.utils._
 object TypedCodecSpec {
   def apply(pt: PType, bufferSpec: BufferSpec): TypedCodecSpec = {
     val eType = EType.defaultFromPType(pt)
-    TypedCodecSpec(eType, pt.virtualType, bufferSpec)
+    TypedCodecSpec(eType, pt, bufferSpec)
   }
 }
 
-final case class TypedCodecSpec(_eType: EType, _vType: Type, _bufferSpec: BufferSpec) extends AbstractTypedCodecSpec {
+final case class TypedCodecSpec(_eType: EType, _pType: PType, _bufferSpec: BufferSpec) extends AbstractTypedCodecSpec {
```

Comment:
Ah, this won't work until I fix the parsing side. If we wish to not do this in this PR, I proposed we keep encodedPhysicalType here, and trivially make it the canonical representation of the virtualType.deepOptional()